### PR TITLE
Fix: a issue with loading e-Reader dot code

### PR DESCRIPTION
**Pull request by s520**
This PR fix a issue with loading e-Reader dot code.

Specifically, it is as follows.
- Even if Dot code file is not specified, it tries to read as eternal and does not accept button input.
- In Japanese etc., the path of Dot code file is garbled internally.

### CODE REVIEW

**Comment by denisfa** [comment]
@s520 Thank you very much for your contribution! I have some questions, though. I will put them on the code.

File: src/gba/GBA.cpp [hunk 1]
```diff
@@ -1712,11 +1712,21 @@ const char* GetSaveDotCodeFile()
     return saveDotCodeFile;
 }
 
+void ResetLoadDotCodeFile()
+{
+    loadDotCodeFile = strdup("");
```
Comment (denisfa) [inline src/gba/GBA.cpp]: @s520 Is this leaking memory? I believe you should `if (loadDotCodeFile) free(loadDotCodeFile)` before the `strdup`.

File: src/gba/GBA.cpp [hunk 2]
```diff
@@ -1712,11 +1712,21 @@ const char* GetSaveDotCodeFile()
...
+}
+
 void SetLoadDotCodeFile(const char* szFile)
 {
     loadDotCodeFile = strdup(szFile);
 }
 
+void ResetSaveDotCodeFile()
+{
+    saveDotCodeFile = strdup("");
```
Comment (denisfa) [inline src/gba/GBA.cpp]: Likewise.

File: src/wx/cmdevents.cpp [hunk 3]
```diff
@@ -869,7 +878,7 @@ EVT_HANDLER_MASK(SaveDotCodeFile, "Save e-Reader Dot Code...", CMDEN_GBA)
         return;
 
     savedotcodefile_path = dlg.GetPath();
-    SetSaveDotCodeFile(savedotcodefile_path.mb_str(wxConvUTF8));
+    SetSaveDotCodeFile(savedotcodefile_path.mb_str());
```
Comment (denisfa) [inline src/wx/cmdevents.cpp]: @s520 Just for clarification, were the issues regarding the path being garbled on Windows?

Comment (s520) [reply src/wx/cmdevents.cpp]: Yes. I confirmed it in Japanese environment of Windows 10 64bit.
I don't know wxWidgets in detail, but if you specify wxConvUTF8 as an argument, this part will be garbled if the path contains Japanese (multibyte characters?).

File: src/gba/GBA.cpp (hunk 1 above)
Comment (s520) [reply src/gba/GBA.cpp]: Oops, that's right.
I will fix it.

Comment (s520) [reply src/gba/GBA.cpp]: Please check that I have fixed it.

File: src/gba/GBA.cpp (hunk 2 above)
Comment (s520) [reply src/gba/GBA.cpp]: I have also fixed this, so please check it.

File: src/wx/cmdevents.cpp (hunk 3 above)
Comment (denisfa) [reply src/wx/cmdevents.cpp]: Thank you! We have some bugs about converting between UTF16 and UTF8 for windows, so I just wanted to confirm this was part of it. Nevertheless, this should be enough for now.

**Pull request merged**